src/module-lookup.ts

```diff
@@ -271,7 +271,15 @@ function getGlobalNpmPaths(filterPaths = true): string[] {
   }
 
   if (process.env.PNPM_HOME) {
-    paths.push(resolve(process.env.PNPM_HOME, 'global/*/node_modules'));
+    const pnpmHome = process.env.PNPM_HOME;
+    paths.push(
+      ...globbySync('global/*/node_modules', {
+        cwd: pnpmHome,
+        onlyDirectories: true,
+        suppressErrors: true,
+        absolute: true,
+      }).map(modulesPath => resolve(pnpmHome, modulesPath)),
```

Comment:
Since we are requesting absolute paths, pnpmHome in resolve should not be required.
```suggestion
      }).map(modulesPath => resolve(modulesPath)),
```